src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => {
+      setInitialTodos(todosFromServer);
+      setTodos(todosFromServer);
+      setIsDataLoaded(true);
+    });
+  }, []);
+
+  useEffect(() => {
+    switch (filter) {
+      case Filter.All:
+        setTodos(initialTodos);
+        break;
+
+      case Filter.Active:
+        setTodos(initialTodos.filter(todo => !todo.completed));
+        break;
+
+      case Filter.Completed:
+        setTodos(initialTodos.filter(todo => todo.completed));
+        break;
+
+      default:
+        break;
+    }
+
+    setTodos(
```

Comment:
it would be bettter to create todos copy, filter it, and only then invoke `setTodos` with correct data